presto-orc/src/main/java/com/facebook/presto/orc/reader/StructStreamReader.java

```diff
@@ -196,6 +184,28 @@ public String toString()
                 .toString();
     }
 
+    private Block[] getBlocksForType(Type type, int positionCount) throws IOException
+    {
+        RowType rowType = (RowType) type;
+
+        Block[] blocks = new Block[rowType.getFields().size()];
+
+        for (int i = 0; i < rowType.getFields().size(); i++) {
+            Optional<String> fieldName = rowType.getFields().get(i).getName();
+            Type fieldType = rowType.getFields().get(i).getType();
+
+            if (fieldName.isPresent() && structFields.containsKey(fieldName.get())) {
```

Comment:
If the field name isn't present, shouldn't that be an error?

Reply:
I guess I don't know under what circumstances this can happen. I thought in case we have mix of fields, some with names and some without, it might be better to read the fields that we can, rather than erroring out.

Reply:
The `type` field is being supplied by Presto, based on the struct field names declared in the metastore, so I would say it is a bug in Presto if the names are not present... so let's throw an exception.